Mathlib/Topology/Category/CompHausLike/SigmaComparison.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2024 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.Data.Countable.Small
+import Mathlib.Topology.Category.CompHausLike.Limits
+/-!
+
+# The sigma-comparison map
+
+This file defines the map `CompHausLike.sigmaComparison` associated to a presheaf `X` on
+`CompHausLike P`, and a finite family `S₁,...,Sₙ` of spaces in `CompHausLike P`, where `P` is
+stable under taking finite disjoint unions.
+
+The map `sigmaComparison` is the canonical map `X(S₁ ⊔ ... ⊔ Sₙ) ⟶ X(S₁) × ... × X(Sₙ)` induced by
+the inclusion maps `Sᵢ ⟶ S₁ ⊔ ... ⊔ Sₙ`, and it is an isomorphism when `X` preserves finite
+products.
+-/
+
+universe u w
+
+open CategoryTheory Limits
+
+namespace CompHausLike
+
+variable {P : TopCat.{u} → Prop} [HasExplicitFiniteCoproducts.{u} P]
+  (X : (CompHausLike.{u} P)ᵒᵖ ⥤ Type max u w) [PreservesFiniteProducts X]
+  {α : Type u} [Finite α] (σ : α → Type u)
+  [∀ a, TopologicalSpace (σ a)] [∀ a, CompactSpace (σ a)] [∀ a, T2Space (σ a)]
+  [∀ a, HasProp P (σ a)]
+
+instance : HasProp P (Σ (a : α), (σ a)) := HasExplicitFiniteCoproducts.hasProp (fun a ↦ of P (σ a))
+
+/--
+The comparison map from the value of a condensed set on a finite coproduct to the product of the
+values on the components.
+-/
+def sigmaComparison : X.obj ⟨(of P ((a : α) × σ a))⟩ ⟶ ((a : α) → X.obj ⟨of P (σ a)⟩) :=
+  fun x a ↦ X.map ⟨Sigma.mk a, continuous_sigmaMk⟩ x
+
+noncomputable instance : PreservesLimitsOfShape (Discrete α) X :=
+  let α' := (Countable.toSmall α).equiv_small.choose
+  let e : α ≃ α' := (Countable.toSmall α).equiv_small.choose_spec.some
+  letI : Fintype α := Fintype.ofFinite _
+  letI : Fintype α' := Fintype.ofEquiv α e
+  preservesLimitsOfShapeOfEquiv (Discrete.equivalence e.symm) X
+
+theorem sigmaComparison_eq_comp_isos : sigmaComparison X σ =
+    (X.mapIso (opCoproductIsoProduct'
+      (finiteCoproduct.isColimit.{u, u} (fun a ↦ of P (σ a)))
+      (productIsProduct fun x ↦ Opposite.op (of P (σ x))))).hom ≫
+    (PreservesProduct.iso X fun a ↦ ⟨of P (σ a)⟩).hom ≫
+    (Types.productIso.{u, max u w} fun a ↦ X.obj ⟨of P (σ a)⟩).hom := by
+  ext x a
+  simp only [Cofan.mk_pt, Fan.mk_pt, Functor.mapIso_hom,
+    PreservesProduct.iso_hom, types_comp_apply, Types.productIso_hom_comp_eval_apply]
+  have := congrFun (piComparison_comp_π X (fun a ↦ ⟨of P (σ a)⟩) a)
+  simp only [types_comp_apply] at this
+  rw [this, ← FunctorToTypes.map_comp_apply]
+  simp only [sigmaComparison]
+  apply congrFun
+  congr 2
+  erw [← opCoproductIsoProduct_inv_comp_ι]
+  simp only [coe_of, Opposite.unop_op, unop_comp, Quiver.Hom.unop_op, Category.assoc]
+  change finiteCoproduct.ι.{u, u} (fun a ↦ of P (σ a)) _ = _
```

Comment:
```suggestion
```
This line does not seem to be necessary for the proof to compile (for me), but maybe there are peformance / readability reasons to include it? Up to you.